### PR TITLE
Separate New column names and transform expressions in TransformSQL

### PR DESCRIPTION
**Pull request by phvu**
### Description and related tickets, documents
- Added transformUDFWithNames to replace transformUDF. Separating names and transformation expressions
- Deprecated transformUDF (in `IHandleTransformations`)

Reviewers: @Huandao0812 @nhanitvn @hai-adatao 

### Breaking changes & backward compatible issues
No

### How to test
UT

### PR Progress
Make sure all checkboxes below are checked before merged
- [x] Branch is in format `prefix/description` (see [this](http://www.guyroutledge.co.uk/blog/git-branch-naming-conventions/))
- [x] Merge check has no conflicts. PR checks passed.
- [x] Code review is done.

### CODE REVIEW

File: core/src/main/java/io/ddf/etl/TransformationHandler.java [hunk 1]
```diff
@@ -222,14 +218,131 @@ public static String RToSqlUdf(List<String> RExps, List<String> selectedColumns,
...
       selectStr += (exp + ",");
     }
 
     return selectStr.substring(0, selectStr.length() - 1);
   }
 
+  /**
+   * Create new columns or overwrite existing ones
+   *
+   * @param newColumnNames       Array of new column names.
```
Comment (nhanitvn) [inline core/src/main/java/io/ddf/etl/TransformationHandler.java]: Should add details on the case of anonymous expressions

Comment (phvu) [inline core/src/main/java/io/ddf/etl/TransformationHandler.java]: done

Comment (Huandao0812) [inline core/src/main/java/io/ddf/etl/TransformationHandler.java]: can we limit the number of column and expression to be one at a time, I don't see many use case where DS need to transform multiple columns at a time.

Comment (phvu) [inline core/src/main/java/io/ddf/etl/TransformationHandler.java]: Maybe no in interactive scenarios, but once they write it up in a script, they may want to pack them all into a single function invoke.
Is there any technical disadvantage with this approach though? We can add wrapper which takes single expression, similar to `transformUDF`.

**Pull request merged**